challenge_16/views/app/show_data.html.erb

```diff
@@ -1,7 +1,10 @@
-<ul>
-<% @blogs.each do |blog| %>
-  <li>
-    <strong>Title: <%= blog.title %> </strong>, Content: <%= blog.content %>
-  </li>
+<%= tag.ul do %>
+  <% @blogs.each do |blog| %>
+    <%= tag.li do %>
+      <%= tag.strong do %>
+        Title: <%= blog.title %>,
```

Comment:
Ah yes, my bad! I noticed that moving it to be outside the block to `tag.strong` was introducing an extra space:
```ruby
    <%= tag.li do %>
      <%= tag.strong do %>
        Title: <%= blog.title %>
      <% end %>,
      Content: <%= blog.content %>
    <% end %>
```
![Screen Shot 2021-05-14 at 8 58 51 AM](https://user-images.githubusercontent.com/22918438/118273995-9bf9db00-b492-11eb-8764-8d31b36c3d2b.png)

So I moved it into the block to experiment and then forgot about it 😅 

Putting the block on a single line with the tag helper fixes this:
```ruby
<%= tag.strong do %>Title: <%= blog.title %><% end %>,
```
![Screen Shot 2021-05-14 at 9 01 51 AM](https://user-images.githubusercontent.com/22918438/118274353-090d7080-b493-11eb-9f84-246bde055753.png)

Is it bad style to have a `do; end` block across one line, or would this be the best way to do it?

Reply:
I fear there’s no perfect solution here. 😢 

Whitespace in ERB is a bit of a pain. The [`ActionView::Base` documentation](https://api.rubyonrails.org/classes/ActionView/Base.html#class-ActionView::Base-label-ERB) says:

>When on a line that only contains whitespaces except for the tag, `<% %>` suppresses leading and trailing whitespace, including the trailing newline. `<% %>` and `<%- -%>` are the same. Note however that `<%= %>` and `<%= -%>` are different: only the latter removes trailing whitespaces.

I think this behaviour comes from [erubi](https://github.com/jeremyevans/erubi) but, either way, it’s neither very reliable nor very well documented. In any case, it does reveal that ERB has some whitespace-suppression features which can be turned on by adding the `-` character inside the tags. So I think this would do what you want…

```erb
<%= tag.ul do %>
  <% @blogs.each do |blog| %>
    <%= tag.li do %>
      <%= tag.strong do %>
        Title: <%= blog.title -%>
<% end %>,
      Content: <%= blog.content %>
    <% end %>
  <% end %>
<% end %>
```

…by suppressing the newline at the end of the `Title:` line (and manually removing the indentation before the following comma), but that’s not very nice!

Your solution of putting the whole `<strong>` tag on one line, thereby avoiding any newlines or indentation inside the tag altogether, is probably the least bad option. I wouldn’t say it’s bad style; it’s not ideal but it’s sometimes necessary for this exact reason.

Reply:
Gotcha! Thanks for digging into the details for me ❤️ In writing ERB, it does often feel like you have to "cheat" a bit in one way or another to get exactly what you want 🤪